Confus/Confus/OpenAL/OpenALListener.cpp

```diff
@@ -0,0 +1,63 @@
+#include <stdexcept>
+
+#include "OpenALListener.h";
+
+namespace Confus {
+    struct Vector3 {
+        float x, y, z;
+    };
+
+    void OpenALListener::init()
+    {
+        OpenAL::ALFWInit();
+        if(!OpenAL::ALFWInitOpenAL())
+        {
+            OpenAL::ALFWShutdown();
+            throw std::invalid_argument("Failed to init");
+        }
+    }
+
+    void OpenALListener::dispose()
+    {
+        OpenAL::ALFWShutdownOpenAL();
+        OpenAL::ALFWShutdown();
+    }
+
+
+    void OpenALListener::setPosition(float a_PositionX, float a_PositionY, float a_PositionZ)
+    {
+        alListener3f(AL_POSITION, a_PositionX, a_PositionY, a_PositionZ);
+    }
+
+    void OpenALListener::setPosition(Vector3 a_Position) {
```

Comment:
He was wondering whether you couldn't assign the vector at once directly, but Lansenou has already overloaded the function to support that. However, that overload uses a custom struct rather than irrlicht's built-in vector, so I'd still change that